src/app/components/AdminEventDetails/AdminEventDetails.tsx

```diff
@@ -0,0 +1,102 @@
+"use client";
+import style from './AdminEventDetails.module.css';
+import AdminEventDetailsButton from './AdminEventDetailsButton';
+import AccessTimeIcon from '@mui/icons-material/AccessTime';
+import LocationOnOutlinedIcon from '@mui/icons-material/LocationOnOutlined';
+import SegmentIcon from '@mui/icons-material/Segment';
+import PeopleOutlineIcon from '@mui/icons-material/PeopleOutline';
+import ImportContactsIcon from '@mui/icons-material/ImportContacts';
+import { useEffect, useState } from 'react';
+import { IEvent } from '@database/eventSchema';
+
+type Props = {
+  _id: string ;
+};
+  
+async function getEvent(_id: string) {
+  try {
+    const res = await fetch(`http://localhost:3000/api/event/${_id}`, { //default GET
+      cache: "no-store", 
+    } );
+
+    if (!res.ok) {
+      throw new Error("Failed to fetch blog");
+    }
+    console.log("Data: " + res.json)
+    return res.json();
+  } catch (err: unknown) {
+    console.log(`error: ${err}`);
+    return null;
+  }
+}
+
+function getDayName(date: Date) {
+  const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+  const eventDay = new Date(date).getDay();
+  const eventDayName = days[eventDay];
+  return eventDayName;
+}
+
+export default function AdminEventDetails({ _id }: Props) {
+  const [event, setEvent] = useState<IEvent | null>(null);
+
+  const tempVolunteerRole = { //temporary volunteer role, once we have the volunteer data, we can remove this
+    startTime: new Date("2024-02-26T08:00:00.000-08:00").toLocaleTimeString('en-US', { hour: '2-digit', minute: '2-digit', hour12: true }),
+    endTime: new Date("2024-02-26T08:00:00.000-09:00").toLocaleTimeString('en-US', { hour: '2-digit', minute: '2-digit', hour12: true }),
+    volunteers: ['volunteer1', 'volunteer2', 'volunteer3'],
+  };
```

Comment:
This data is already present in each volunteerRole document. Look at the schema, you'll see that each volunteerRole has a roleName, description, event(id), and timeslots. timeslots is an array of timeslots, which are objects (as you have here) of starttime, endtime, and array of volunteer ids.